### PR TITLE
✨Trinity TTS player amp-trinity-tts-player extension

### PR DESCRIPTION
**Pull request by industral**
Implemented https://github.com/ampproject/amphtml/issues/26836

### CODE REVIEW

**Comment by industral** [comment]
@googlebot I signed it!

**Comment by industral** [comment]
I can see flickering tests, each time fails different one, not related to that extension.

**Comment by industral** [comment]
@amaltas @jridgewell @ampproject/wg-ui-and-a11y can you guys help me, can't pass PR for 2 weeks now, as it fails every time with different issues not related to my code, e.g. checking, e2e tests, visual checking, etc.. each time new failing. I can't rerun them, so have to wait when a new code arrives into master and make remove merge/push in order to run checking again...
Maybe I'm doing something wrong, and I shouldn't use master branch? So far it's broken.
Please let me know how I can pass it through. Thanks!

**Comment by jridgewell** [comment]
The visual diff tests seems flaky, you might need to rebase off current master to fix it. We can manually approve if not.

But this still needs a review from @ampproject/wg-ui-and-a11y

**Comment by industral** [comment]
so far visual fails each time on that one https://percy.io/ampproject/amphtml/builds/4388616?utm_campaign=ampproject&utm_content=amphtml&utm_source=github_status_public

**Comment by industral** [comment]
@jridgewell now it fails on other components https://percy.io/ampproject/amphtml/builds/4425094?utm_campaign=ampproject&utm_content=amphtml&utm_source=github_status_public (facepalm)

**Comment by industral** [comment]
visual still failing https://percy.io/ampproject/amphtml/builds/4430937?utm_campaign=ampproject&utm_content=amphtml&utm_source=github_status_public. @jridgewell can you please take a look or accept it manually?

**Comment by industral** [comment]
now it failing on cicd on some other component

**Comment by industral** [comment]
@jridgewell finally I see all tests passed 👍But I see `ampproject/owners-check — Missing required OWNERS approvals!`. Who should approve it? Is there any required actions from my side?

File: extensions/amp-trinity-tts-player/0.1/amp-trinity-tts-player.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      iframe.src = src;
+
+      this.applyFillContent(iframe);
+      this.element.appendChild(iframe);
+    });
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    const placeholder = this.win.document.createElement('amp-img');
```
Comment (alanorozco) [inline extensions/amp-trinity-tts-player/0.1/amp-trinity-tts-player.js]: We prefer standard loaders across AMP components. 

1. Remove `createPlaceholderCallback()`.
2. Implement `createLoaderLogoCallback()` if you'd like a custom logo that is **contained** in a loader. Otherwise just remove `createLoaderLogoCallback()`.

File: build-system/tasks/check-links.js [hunk 2]
```diff
@@ -134,7 +134,8 @@ function checkLinksInFile(file) {
         return;
       }
       let containsDeadLinks = false;
-      for (const {link, status, statusCode} of results) {
+      // eslint-disable-next-line prefer-const
+      for (let {link, status, statusCode} of results) {
```
Comment (alanorozco) [inline build-system/tasks/check-links.js]: Please undo this change.

File: extensions/amp-trinity-tts-player/0.1/amp-trinity-tts-player.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        }
+      });
+
+      iframe.addEventListener('load', () => {
+        iframe.contentWindow./*OK*/ postMessage(
+          /** @type {JsonObject} */ ({
+            type: 'init',
+            data: {
+              text: this.win.document.body./*OK*/ innerHTML,
+              campaignId: this.campaignId_,
```
Comment (alanorozco) [inline extensions/amp-trinity-tts-player/0.1/amp-trinity-tts-player.js]: This is all the messaging the component does.

It comes from one parameter, defined once and never changing.

Why can't this be a URL parameter on an `amp-iframe`? Could you potentially host a page that would allow your users to embed this on any iframe like `/player/trinity-amp?campaignId=FOO`?

You could also use `amp-video-iframe` to set it as a data param for author ergonomics:

```html
<amp-video-iframe 
  src="https://trinitymedia.ai/player/trinity-amp"
  data-param-campaign-id="FOO"
></amp-video-iframe>
```

It would accomplish the same.

**Comment by industral** [comment]
Hi @alanorozco!
Thanks for taking this review! I'm in GMT+2 TZ, but will try to fix it ASAP in order to pass it thought!

File: extensions/amp-trinity-tts-player/0.1/amp-trinity-tts-player.js (hunk 3 above)
Comment (industral) [reply extensions/amp-trinity-tts-player/0.1/amp-trinity-tts-player.js]: or you talking specifically to remove passing `campaignId`?

Comment (alanorozco) [reply extensions/amp-trinity-tts-player/0.1/amp-trinity-tts-player.js]: Ah, just noticed the `innerHTML` reference.

I'm not sure passing all of the page's content to a third-party without user-consent follows our privacy principles.

I've copied `wg-security-privacy` on [the I2I](https://github.com/ampproject/amphtml/issues/26836) for approval. I will not make any more reviews to this PR until that's approved. Please let's continue discussion there :)

Comment (industral) [reply extensions/amp-trinity-tts-player/0.1/amp-trinity-tts-player.js]: OK, thanks @alanorozco ! Let's wait for their approval or suggesting how we should handle that in other way.

**Comment by alanorozco** [comment]
@industral Thanks for clarifying. It seems that the use case for these requirements is passing the page's content to the external service, which we need to discuss first. As mentioned in above comment, please follow I2I.

File: build-system/tasks/check-links.js (hunk 2 above)
Comment (alanorozco) [reply build-system/tasks/check-links.js]: It's a legitimate problem, thanks for bringing it to my attention.

I'm fixing it individually in #27197.

**Comment by mrjoro** [comment]
Is this PR still active?

**Comment by industral** [comment]
> Is this PR still active?

nope, we can close. We can't achieve what we want with current provided functionality by AMP :/
